src/directwrite/mod.rs

```diff
@@ -238,12 +232,33 @@ impl crate::Rasterize for DirectWriteRasterizer {
     fn get_glyph(&mut self, glyph: GlyphKey) -> Result<RasterizedGlyph, Error> {
         let loaded_font = self.get_loaded_font(glyph.font_key)?;
 
-        match self.rasterize_glyph(&loaded_font.face, glyph.size, glyph.c) {
-            Err(err @ Error::MissingGlyph(_)) => {
-                let fallback_font = self.get_fallback_font(&loaded_font, glyph.c).ok_or(err)?;
-                self.rasterize_glyph(&fallback_font.create_font_face(), glyph.size, glyph.c)
+        let mut glyph_index = self.get_glyph_index(&loaded_font.face, glyph.c);
+        let fallback_font;
+
+        let font = match glyph_index {
+            Some(_) => loaded_font,
+            None => match self.get_fallback_font(&loaded_font, glyph.c) {
+                None => loaded_font,
+                Some(font) => {
+                    fallback_font = Font::from(font);
+                    glyph_index = self.get_glyph_index(&fallback_font.face, glyph.c);
+
+                    &fallback_font
+                },
             },
-            result => result,
+        };
```

Comment:
This seems like it would be something that could also be cleaned up a bit. If the other suggestion doesn't help with doing so please let me know and I'll have to play around with it myself.

Reply:
The problem is that you should create a font in some cases, which is not the same as on macOS with similar construction, though might work.